lib/screens/search/search_results_page.dart

```diff
@@ -30,22 +35,34 @@ class _SearchResultsPageState extends ConsumerState<SearchResultsPage> {
     if (widget.searchInput != '') {
       setState(() {
         _resultFuture = debouncer.run(
-          () => client
-              .query$Search(
-            Options$Query$Search(
-              variables: Variables$Query$Search(queryString: widget.searchInput),
-            ),
-          )
-              .onError((error, stackTrace) {
-            throw Error();
-          }).then(
-            (value) {
-              if (value.hasException) {
-                throw ErrorDescription(value.exception.toString());
-              }
-              return value.parsedData?.search;
-            },
-          ),
+          () {
+            final searchStopwatch = Stopwatch()..start();
+            return client
+                .query$Search(
+              Options$Query$Search(
+                variables: Variables$Query$Search(queryString: widget.searchInput),
+              ),
+            )
+                .onError((error, stackTrace) {
+              throw Error();
+            }).then(
+              (value) {
+                searchStopwatch.stop();
+                if (value.hasException) {
+                  throw ErrorDescription(value.exception.toString());
+                }
+                final searchResult = value.parsedData?.search;
+                if (searchResult != null) {
+                  ref.read(analyticsProvider).searchPerformed(SearchPerformedEvent(
```

Comment:
It's a better idea to do the analytics on a seperate _resultFuture.then() call.
If for some reason the analytics code throws an exception, the whole search will fail.